Expensify/App
# Download E.cash attachments immediately

### Details
Passes an auth token in the chat attachment URL, if necessary.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/154254

### Tests / QA Steps
1. Send a regular link such as https://www.google.com/ in Expensify.cash
2. Click on the link, ensure that it is opened in the default browser.
3. Click on `Upload Attachment` and upload a `.zip` file.
4. After a few moments, a link to that attachment should appear.
5. Click on that link, and ensure that the file is downloaded immediately. You should not have to log in to the expensify.com website.

### Tested On


### Screenshots
#### Web
https://user-images.githubusercontent.com/47436092/119179677-117e2200-ba24-11eb-955e-147b6efe4329.mov

#### Mobile Web
https://user-images.githubusercontent.com/47436092/119179266-93218000-ba23-11eb-9b4f-8ba35d96c229.mov

#### Desktop
https://user-images.githubusercontent.com/47436092/119179975-79cd0380-ba24-11eb-8f5d-7660592cd459.mov

#### iOS
https://user-images.githubusercontent.com/47436092/119180502-3a52e700-ba25-11eb-9977-71d71c9a2516.mov

#### Android
